templates/commands/goldentest/new_test_cli.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package goldentest implements golden test related subcommands.
+package goldentest
+
+// This file implements the "templates golden-test new-test" subcommand.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/abcxyz/abc/templates/common"
+	"github.com/abcxyz/abc/templates/common/input"
+	"github.com/abcxyz/abc/templates/common/specutil"
+	"github.com/abcxyz/abc/templates/common/templatesource"
+	"github.com/abcxyz/abc/templates/model/decode"
+	goldentest "github.com/abcxyz/abc/templates/model/goldentest/v1beta3"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+)
+
+type NewTestCommand struct {
+	cli.BaseCommand
+
+	flags NewTestFlags
+}
+
+func (c *NewTestCommand) Desc() string {
+	return "create a new golden test"
+}
+
+func (c *NewTestCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options] <test_name> <location>
+
+The {{ COMMAND }} create a new golden test.
+
+The "<test_name>" is the name of the test.
+
+The "<location>" is the location of the template.
+`
+}
+
+func (c *NewTestCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+	c.flags.Register(set)
+	return set
+}
+
+func (c *NewTestCommand) Run(ctx context.Context, args []string) (rErr error) {
+	logger := logging.FromContext(ctx)
+
+	if err := c.Flags().Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	tempDir, err := os.MkdirTemp("", "abc-new-test-*")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory: %w", err)
+	}
+	defer func() {
+		rErr = errors.Join(rErr, os.RemoveAll(tempDir))
+	}()
+
+	wd, err := c.WorkingDir()
+	if err != nil {
+		return fmt.Errorf("failed to get working directory: %w", err)
+	}
+
+	fs := &common.RealFS{}
+	logger.DebugContext(ctx, "downloading/copying template")
+	_, templateDir, err := templatesource.Download(ctx, &templatesource.DownloadParams{
```

Comment:
I think maybe the call to Download() is unnecessary. Since the template is already required to be in a local directory, we can load the spec file from that directory without needing to do a download.

Reply:
Yep, will remove it.